internal/testing/factories/factories.go

```diff
@@ -0,0 +1,50 @@
+package factories
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"golang.org/x/crypto/ssh"
+)
+
+var RSAGenerationError = errors.New("rsa key generation failed")
+
+var nameSequence = 1
```

Comment:
Although we do not have parallel tests right now, I would definitely encourage to do defensive programming and use atomic integer here. I am all for running unit tests in parallel as soon as possible so we avoid bigger refactoring later.